### PR TITLE
Add in native variants using Object and Map

### PR DESCRIPTION
**Pull request by blowery**
This PR adds in native variants of the tests, just to see how hamt-style structures compare.

Also, update dependencies.

### CODE REVIEW

**Comment by mattbierner** [comment]
Thank you for looking into this. I like the idea, but have some concerns about how a few of the new benchmarks are currently written: Put_all and remove_all specifically (the transient versionslook ok at a glance).

To emulate what the other libraries are benchmarking, the map should be copied before each update. That way, every temporary map object is immutable as well. For the transient case, this doesn't matter since we only care about the first and end results, but immutability is the whole point of the non-transient case. 

Remove_all is also benchmarking the wrong thing currently. The result of remove_all is just an empty map, but the benchmark should be the cost to remove every element from the map (immutably) one at a time. 

If you can address these concerns I'd be happy to merge this in. Also, I'd love to see some initial benchmark results for the new cases of you have them. I'm on the road so it may take me a little while to respond. Sorry I can't provide more detailed feedback at the moment.

**Comment by blowery** [comment]
Thanks for the quick look @mattbierner! 

> To emulate what the other libraries are benchmarking, the map should be copied before each update. That way, every temporary map object is immutable as well. For the transient case, this doesn't matter since we only care about the first and end results, but immutability is the whole point of the non-transient case.

> Remove_all is also benchmarking the wrong thing currently. The result of remove_all is just an empty map, but the benchmark should be the cost to remove every element from the map (immutably) one at a time.

Makes sense. I wasn't sure exactly which implementation made sense, since with native you'd never really do the non-transient version for a method named `removeAll`, but if the intention is to bench removing each item, one at a time, that makes perfect sense. I'll update the test and repush.

Would you prefer a new commit on top of dd9151a or would you like me to replace it with the proper one changes?

**Comment by blowery** [comment]
FYI, the main motivation for adding these tests is to explore how the new ES7 bits that some folks are using to make up "immutable" objects stack up. See https://egghead.io/lessons/javascript-redux-avoiding-array-mutations-with-concat-slice-and-spread?series=getting-started-with-redux and https://egghead.io/lessons/javascript-redux-avoiding-object-mutations-with-object-assign-and-spread?series=getting-started-with-redux

**Comment by blowery** [comment]
Changes all pushed. Here are the numbers I'm seeing:
on: node 4.2.3, mac osx 10.11.2, 3.5 GHz Intel Core i5

```
Get nth
nativeObject(10)              :      8047493.27 +/- 0.40% op/s
nativeMap(10)                 :     21259060.99 +/- 0.48% op/s
hashtrie(10)                  :     11497715.60 +/- 2.43% op/s
hamt(10)                      :     12730477.00 +/- 0.50% op/s
hamt_plus(10)                 :     12630066.44 +/- 0.60% op/s
persistent-hash-trie(10)      :      9181959.16 +/- 0.42% op/s
mori hash_map(10)             :      2399847.05 +/- 3.23% op/s
immutable(10)                 :     11639506.12 +/- 0.46% op/s
nativeObject(100)             :     14707367.72 +/- 0.50% op/s
nativeMap(100)                :     19112508.12 +/- 0.50% op/s
hashtrie(100)                 :     10694389.83 +/- 0.51% op/s
hamt(100)                     :     12060938.22 +/- 0.43% op/s
hamt_plus(100)                :     12606433.66 +/- 0.70% op/s
persistent-hash-trie(100)     :      3863398.91 +/- 0.37% op/s
mori hash_map(100)            :      2616905.62 +/- 0.36% op/s
immutable(100)                :     11199596.72 +/- 0.41% op/s
nativeObject(1000)            :     14495388.42 +/- 0.51% op/s
nativeMap(1000)               :     16911117.30 +/- 0.40% op/s
hashtrie(1000)                :      8587511.92 +/- 0.38% op/s
hamt(1000)                    :      9911445.63 +/- 0.44% op/s
hamt_plus(1000)               :      9538066.77 +/- 0.38% op/s
persistent-hash-trie(1000)    :      1993496.17 +/- 0.34% op/s
mori hash_map(1000)           :       858882.95 +/- 0.41% op/s
immutable(1000)               :      8796403.78 +/- 0.37% op/s
nativeObject(10000)           :      9750157.44 +/- 0.44% op/s
nativeMap(10000)              :     13597384.02 +/- 0.62% op/s
hashtrie(10000)               :      5202128.47 +/- 2.84% op/s
hamt(10000)                   :      7151111.31 +/- 0.54% op/s
hamt_plus(10000)              :      6549176.37 +/- 0.53% op/s
persistent-hash-trie(10000)   :      1374079.20 +/- 0.94% op/s
mori hash_map(10000)          :       658498.79 +/- 5.14% op/s
immutable(10000)              :      5612141.90 +/- 0.67% op/s
nativeObject(100000)          :      2597926.44 +/- 1.22% op/s
nativeMap(100000)             :      3160015.05 +/- 2.48% op/s
hashtrie(100000)              :      1436749.96 +/- 1.21% op/s
hamt(100000)                  :      1732702.92 +/- 0.97% op/s
hamt_plus(100000)             :      1643812.32 +/- 1.02% op/s
persistent-hash-trie(100000)  :       970316.81 +/- 0.57% op/s
mori hash_map(100000)         :       515264.38 +/- 0.50% op/s
immutable(100000)             :      1175641.69 +/- 0.74% op/s


put nth
nativeObject(10)              :        92414.41 +/- 22.96% op/s
nativeMap(10)                 :       613684.27 +/- 0.33% op/s
hashtrie(10)                  :      4900805.74 +/- 0.30% op/s
hamt(10)                      :      7069688.60 +/- 0.46% op/s
hamt_plus(10)                 :      2769501.12 +/- 0.39% op/s
persistent-hash-trie(10)      :       347530.23 +/- 0.61% op/s
mori hash_map(10)             :       774126.82 +/- 0.42% op/s
immutable(10)                 :      3491514.55 +/- 0.43% op/s
nativeObject(100)             :        32570.63 +/- 0.32% op/s
nativeMap(100)                :        70868.66 +/- 0.36% op/s
hashtrie(100)                 :      2829822.87 +/- 0.39% op/s
hamt(100)                     :      2588307.93 +/- 0.38% op/s
hamt_plus(100)                :      1741995.90 +/- 0.36% op/s
persistent-hash-trie(100)     :        79794.17 +/- 4.85% op/s
mori hash_map(100)            :      1851540.80 +/- 0.33% op/s
immutable(100)                :      1950466.86 +/- 0.32% op/s
nativeObject(1000)            :         2681.06 +/- 0.38% op/s
nativeMap(1000)               :         7284.85 +/- 0.31% op/s
hashtrie(1000)                :      1474953.81 +/- 0.31% op/s
hamt(1000)                    :      1622148.97 +/- 0.35% op/s
hamt_plus(1000)               :      1315704.31 +/- 0.28% op/s
persistent-hash-trie(1000)    :        69022.51 +/- 0.33% op/s
mori hash_map(1000)           :      1618576.28 +/- 0.29% op/s
immutable(1000)               :      1376308.20 +/- 0.39% op/s
nativeObject(10000)           :          277.79 +/- 0.35% op/s
nativeMap(10000)              :          605.92 +/- 0.48% op/s
hashtrie(10000)               :      1350205.13 +/- 0.29% op/s
hamt(10000)                   :      1103965.57 +/- 10.64% op/s
hamt_plus(10000)              :      1023004.04 +/- 17.87% op/s
persistent-hash-trie(10000)   :        60921.65 +/- 0.28% op/s
mori hash_map(10000)          :      1463894.32 +/- 0.26% op/s
immutable(10000)              :      1110375.03 +/- 7.42% op/s
nativeObject(100000)          :           16.60 +/- 5.83% op/s
nativeMap(100000)             :           32.52 +/- 13.41% op/s
hashtrie(100000)              :      1262930.90 +/- 0.34% op/s
hamt(100000)                  :      1156496.76 +/- 0.28% op/s
hamt_plus(100000)             :      1023376.79 +/- 0.26% op/s
persistent-hash-trie(100000)  :        32372.89 +/- 0.33% op/s
mori hash_map(100000)         :      1292841.37 +/- 0.33% op/s
immutable(100000)             :      1019026.09 +/- 0.31% op/s


Put All
nativeObject(10)              :        68480.14 +/- 0.38% op/s
nativeMap(10)                 :       121714.33 +/- 0.45% op/s
hashtrie(10)                  :       458072.41 +/- 0.45% op/s
hamt(10)                      :       608855.29 +/- 4.93% op/s
hamt_plus(10)                 :       265073.36 +/- 0.41% op/s
persistent-hash-trie(10)      :        52776.06 +/- 0.33% op/s
mori hash_map(10)             :       163016.34 +/- 2.64% op/s
immutable(10)                 :       197950.72 +/- 5.15% op/s
nativeObject(100)             :          634.33 +/- 2.46% op/s
nativeMap(100)                :         1268.96 +/- 6.89% op/s
hashtrie(100)                 :        23927.45 +/- 3.09% op/s
hamt(100)                     :        24455.53 +/- 5.67% op/s
hamt_plus(100)                :        22021.20 +/- 2.08% op/s
persistent-hash-trie(100)     :         1106.31 +/- 2.05% op/s
mori hash_map(100)            :        13853.32 +/- 3.18% op/s
immutable(100)                :        19112.18 +/- 1.85% op/s
nativeObject(1000)            :            5.36 +/- 2.16% op/s
nativeMap(1000)               :           16.05 +/- 1.53% op/s
hashtrie(1000)                :         1685.70 +/- 2.20% op/s
hamt(1000)                    :         1881.42 +/- 2.42% op/s
hamt_plus(1000)               :         1653.66 +/- 1.20% op/s
persistent-hash-trie(1000)    :           65.75 +/- 2.41% op/s
mori hash_map(1000)           :          358.50 +/- 6.91% op/s
immutable(1000)               :         1371.42 +/- 3.17% op/s
nativeObject(10000)           :            0.05 +/- 3.01% op/s
nativeMap(10000)              :            0.15 +/- 0.25% op/s
hashtrie(10000)               :          121.78 +/- 1.46% op/s
hamt(10000)                   :          133.29 +/- 0.81% op/s
hamt_plus(10000)              :          115.51 +/- 0.84% op/s
persistent-hash-trie(10000)   :            4.00 +/- 2.05% op/s
mori hash_map(10000)          :           45.61 +/- 1.63% op/s
immutable(10000)              :          112.05 +/- 1.14% op/s


Put All (transient)
nativeObject(10)              :       403617.06 +/- 2.58% op/s
nativeMap(10)                 :      1282412.86 +/- 1.06% op/s
hamt(10)                      :      1247420.68 +/- 0.78% op/s
hamt_plus(10)                 :       597777.51 +/- 0.33% op/s
mori hash_map(10)             :       197372.73 +/- 0.42% op/s
immutable(10)                 :       299373.34 +/- 0.50% op/s
nativeObject(100)             :        46366.10 +/- 1.41% op/s
nativeMap(100)                :       176720.42 +/- 0.96% op/s
hamt(100)                     :        36683.38 +/- 1.12% op/s
hamt_plus(100)                :        49643.36 +/- 1.05% op/s
mori hash_map(100)            :        22693.49 +/- 1.05% op/s
immutable(100)                :        51581.23 +/- 1.16% op/s
nativeObject(1000)            :         3582.78 +/- 1.76% op/s
nativeMap(1000)               :        19159.32 +/- 1.17% op/s
hamt(1000)                    :         2375.68 +/- 0.81% op/s
hamt_plus(1000)               :         4375.82 +/- 1.06% op/s
mori hash_map(1000)           :          466.93 +/- 1.69% op/s
immutable(1000)               :         4738.48 +/- 1.28% op/s
nativeObject(10000)           :          350.55 +/- 1.17% op/s
nativeMap(10000)              :         1488.58 +/- 1.10% op/s
hamt(10000)                   :          169.38 +/- 0.91% op/s
hamt_plus(10000)              :          334.93 +/- 1.51% op/s
mori hash_map(10000)          :           54.62 +/- 2.47% op/s
immutable(10000)              :          414.06 +/- 1.37% op/s


remove nth
nativeObject(10)              :       422256.67 +/- 0.48% op/s
nativeMap(10)                 :       579560.61 +/- 0.24% op/s
hashtrie(10)                  :      3351925.31 +/- 7.05% op/s
hamt(10)                      :      5647850.66 +/- 0.38% op/s
hamt_plus(10)                 :      2798840.74 +/- 0.28% op/s
persistent-hash-trie(10)      :       197773.32 +/- 0.48% op/s
mori hash_map(10)             :      1867289.77 +/- 0.79% op/s
immutable(10)                 :      2694026.29 +/- 0.95% op/s
nativeObject(100)             :        21334.95 +/- 12.14% op/s
nativeMap(100)                :        75089.39 +/- 0.22% op/s
hashtrie(100)                 :      2317210.23 +/- 0.49% op/s
hamt(100)                     :      2570226.00 +/- 0.55% op/s
hamt_plus(100)                :      1989746.56 +/- 0.41% op/s
persistent-hash-trie(100)     :        64397.33 +/- 0.30% op/s
mori hash_map(100)            :      1446024.17 +/- 0.61% op/s
immutable(100)                :      1761472.26 +/- 0.47% op/s
nativeObject(1000)            :         2244.57 +/- 0.68% op/s
nativeMap(1000)               :         7838.65 +/- 0.51% op/s
hashtrie(1000)                :      1651869.60 +/- 0.29% op/s
hamt(1000)                    :      1698754.69 +/- 0.29% op/s
hamt_plus(1000)               :      1666672.33 +/- 0.47% op/s
persistent-hash-trie(1000)    :        45196.98 +/- 0.88% op/s
mori hash_map(1000)           :       608664.01 +/- 0.34% op/s
immutable(1000)               :      1275972.02 +/- 0.41% op/s
nativeObject(10000)           :          280.86 +/- 0.38% op/s
nativeMap(10000)              :          609.44 +/- 7.61% op/s
hashtrie(10000)               :      1082955.07 +/- 1.01% op/s
hamt(10000)                   :      1233645.34 +/- 0.66% op/s
hamt_plus(10000)              :       950591.54 +/- 1.39% op/s
persistent-hash-trie(10000)   :        25191.48 +/- 0.70% op/s
mori hash_map(10000)          :       484253.63 +/- 0.57% op/s
immutable(10000)              :       886765.17 +/- 1.12% op/s
nativeObject(100000)          :           15.00 +/- 13.09% op/s
nativeMap(100000)             :           38.62 +/- 10.04% op/s
hashtrie(100000)              :       619066.01 +/- 0.50% op/s
hamt(100000)                  :       588381.71 +/- 0.42% op/s
hamt_plus(100000)             :       598449.43 +/- 0.63% op/s
persistent-hash-trie(100000)  :         9924.09 +/- 4.14% op/s
mori hash_map(100000)         :       354322.23 +/- 0.67% op/s
immutable(100000)             :       535818.80 +/- 0.70% op/s


Remove All
nativeObject(10)              :        20083.98 +/- 32.82% op/s
nativeMap(10)                 :        60717.44 +/- 0.30% op/s
hashtrie(10)                  :       397814.34 +/- 0.49% op/s
hamt(10)                      :       835629.90 +/- 0.64% op/s
hamt_plus(10)                 :       372081.47 +/- 0.30% op/s
persistent-hash-trie(10)      :        33592.53 +/- 0.58% op/s
mori hash_map(10)             :       200607.80 +/- 0.44% op/s
immutable(10)                 :       335719.86 +/- 1.42% op/s
nativeObject(100)             :          325.73 +/- 2.70% op/s
nativeMap(100)                :          751.25 +/- 1.93% op/s
hashtrie(100)                 :        24439.78 +/- 1.70% op/s
hamt(100)                     :            0.00 +/- 0.00% op/s
hamt_plus(100)                :        24073.06 +/- 1.41% op/s
persistent-hash-trie(100)     :         3484.22 +/- 1.71% op/s
mori hash_map(100)            :        14374.68 +/- 2.30% op/s
immutable(100)                :        19101.44 +/- 1.82% op/s
nativeObject(1000)            :            2.59 +/- 3.27% op/s
nativeMap(1000)               :            8.12 +/- 0.42% op/s
hashtrie(1000)                :         1600.92 +/- 2.67% op/s
hamt(1000)                    :         1641.41 +/- 2.46% op/s
hamt_plus(1000)               :         2097.90 +/- 2.14% op/s
persistent-hash-trie(1000)    :         1687.20 +/- 1.49% op/s
mori hash_map(1000)           :          543.48 +/- 2.62% op/s
immutable(1000)               :         1414.79 +/- 2.26% op/s
nativeObject(10000)           :            0.03 +/- 3.48% op/s
nativeMap(10000)              :            0.06 +/- 3.46% op/s
hashtrie(10000)               :           84.33 +/- 2.73% op/s
hamt(10000)                   :           95.89 +/- 2.57% op/s
hamt_plus(10000)              :          112.86 +/- 2.52% op/s
persistent-hash-trie(10000)   :          547.42 +/- 2.61% op/s
mori hash_map(10000)          :           43.43 +/- 1.38% op/s
immutable(10000)              :           88.28 +/- 1.54% op/s


Remove All (transient)
nativeObject(10)              :       437654.33 +/- 0.86% op/s
nativeMap(10)                 :       432345.54 +/- 4.49% op/s
hamt(10)                      :       754669.10 +/- 2.90% op/s
hamt_plus(10)                 :       448671.68 +/- 1.79% op/s
mori hash_map(10)             :       146048.09 +/- 2.58% op/s
immutable(10)                 :       592262.43 +/- 2.20% op/s
nativeObject(100)             :        26352.13 +/- 1.54% op/s
nativeMap(100)                :        60219.08 +/- 1.71% op/s
hamt(100)                     :            0.00 +/- 0.00% op/s
hamt_plus(100)                :        45619.58 +/- 1.64% op/s
mori hash_map(100)            :        14627.87 +/- 1.78% op/s
immutable(100)                :        51045.16 +/- 1.99% op/s
nativeObject(1000)            :         2017.67 +/- 8.95% op/s
nativeMap(1000)               :         5591.27 +/- 3.94% op/s
hamt(1000)                    :         1516.04 +/- 3.60% op/s
hamt_plus(1000)               :         4484.86 +/- 1.97% op/s
mori hash_map(1000)           :          624.03 +/- 1.79% op/s
immutable(1000)               :         5017.45 +/- 1.76% op/s
nativeObject(10000)           :          236.55 +/- 1.36% op/s
nativeMap(10000)              :          510.49 +/- 1.92% op/s
hamt(10000)                   :          101.23 +/- 2.22% op/s
hamt_plus(10000)              :          272.48 +/- 2.29% op/s
mori hash_map(10000)          :           52.45 +/- 2.65% op/s
immutable(10000)              :          259.83 +/- 7.99% op/s


Count
nativeObject(10)              :      5399131.18 +/- 1.91% op/s
nativeMap(10)                 :     75319307.88 +/- 0.72% op/s
hashtrie(10)                  :      3194023.19 +/- 0.57% op/s
hamt(10)                      :     14586395.17 +/- 0.59% op/s
hamt_plus(10)                 :     10245588.65 +/- 0.68% op/s
persistent-hash-trie(10)      :       226197.89 +/- 2.59% op/s
mori hash_map(10)             :     82154333.99 +/- 1.80% op/s
immutable(10)                 :    101710850.56 +/- 1.72% op/s
nativeObject(100)             :       346603.34 +/- 3.55% op/s
nativeMap(100)                :     74920856.55 +/- 1.35% op/s
hashtrie(100)                 :            0.00 +/- 0.00% op/s
hamt(100)                     :      1370194.61 +/- 0.86% op/s
hamt_plus(100)                :      1047131.35 +/- 0.37% op/s
persistent-hash-trie(100)     :        12049.55 +/- 1.19% op/s
mori hash_map(100)            :     84234149.26 +/- 1.29% op/s
immutable(100)                :     87976757.07 +/- 4.64% op/s
nativeObject(1000)            :        13231.90 +/- 2.27% op/s
nativeMap(1000)               :     74198361.69 +/- 1.29% op/s
hashtrie(1000)                :            0.00 +/- 0.00% op/s
hamt(1000)                    :        87602.51 +/- 1.91% op/s
hamt_plus(1000)               :        56128.30 +/- 1.45% op/s
persistent-hash-trie(1000)    :          936.76 +/- 6.38% op/s
mori hash_map(1000)           :     81025425.53 +/- 2.28% op/s
immutable(1000)               :     95979930.64 +/- 3.68% op/s
nativeObject(10000)           :          913.72 +/- 2.77% op/s
nativeMap(10000)              :     66481911.96 +/- 0.96% op/s
hashtrie(10000)               :            0.00 +/- 0.00% op/s
hamt(10000)                   :        12282.28 +/- 0.87% op/s
hamt_plus(10000)              :         8769.54 +/- 0.85% op/s
persistent-hash-trie(10000)   :          120.84 +/- 5.67% op/s
mori hash_map(10000)          :     84273110.72 +/- 1.34% op/s
immutable(10000)              :    103312666.24 +/- 1.32% op/s


Sum
nativeObject(10)              :      1293095.97 +/- 0.93% op/s
nativeMap(10)                 :      2298601.08 +/- 2.17% op/s
hashtrie(10)                  :            0.00 +/- 0.00% op/s
hamt(10)                      :      6835885.24 +/- 0.67% op/s
hamt_plus(10)                 :      6997005.41 +/- 0.75% op/s
persistent-hash-trie(10)      :       358786.21 +/- 4.31% op/s
mori hash_map(10)             :      2732311.16 +/- 0.86% op/s
immutable(10)                 :      1613560.35 +/- 3.36% op/s
nativeObject(100)             :        59188.99 +/- 1.19% op/s
nativeMap(100)                :       324391.05 +/- 2.39% op/s
hashtrie(100)                 :            0.00 +/- 0.00% op/s
hamt(100)                     :       616804.31 +/- 0.81% op/s
hamt_plus(100)                :       629266.28 +/- 2.49% op/s
persistent-hash-trie(100)     :         9696.34 +/- 2.54% op/s
mori hash_map(100)            :       373377.58 +/- 1.99% op/s
immutable(100)                :       195877.77 +/- 1.58% op/s
nativeObject(1000)            :         4586.53 +/- 2.08% op/s
nativeMap(1000)               :        29802.57 +/- 3.06% op/s
hashtrie(1000)                :            0.00 +/- 0.00% op/s
hamt(1000)                    :        47678.20 +/- 1.61% op/s
hamt_plus(1000)               :        46015.13 +/- 0.62% op/s
persistent-hash-trie(1000)    :         1129.97 +/- 1.46% op/s
mori hash_map(1000)           :        26479.29 +/- 5.72% op/s
immutable(1000)               :        16902.69 +/- 2.39% op/s
nativeObject(10000)           :          402.70 +/- 2.83% op/s
nativeMap(10000)              :         2831.73 +/- 4.93% op/s
hashtrie(10000)               :            0.00 +/- 0.00% op/s
hamt(10000)                   :         5812.88 +/- 5.15% op/s
hamt_plus(10000)              :         6449.35 +/- 2.81% op/s
persistent-hash-trie(10000)   :          126.01 +/- 5.43% op/s
mori hash_map(10000)          :         3618.99 +/- 3.04% op/s
immutable(10000)              :         2044.14 +/- 1.07% op/s


Keys
nativeObject(10)              :      4886543.05 +/- 6.62% op/s
nativeMap(10)                 :       542581.68 +/- 1.44% op/s
hashtrie(10)                  :      1615988.83 +/- 1.25% op/s
hamt(10)                      :       383231.46 +/- 9.91% op/s
hamt_plus(10)                 :      5320530.94 +/- 2.13% op/s
persistent-hash-trie(10)      :       251776.71 +/- 0.97% op/s
mori hash_map(10)             :       460536.48 +/- 4.87% op/s
immutable(10)                 :       401785.32 +/- 4.94% op/s
nativeObject(100)             :       353286.78 +/- 1.16% op/s
nativeMap(100)                :        60990.71 +/- 2.26% op/s
hashtrie(100)                 :            0.00 +/- 0.00% op/s
hamt(100)                     :        53336.59 +/- 5.25% op/s
hamt_plus(100)                :       502804.48 +/- 2.25% op/s
persistent-hash-trie(100)     :        10427.91 +/- 2.76% op/s
mori hash_map(100)            :        40167.83 +/- 3.20% op/s
immutable(100)                :        89987.36 +/- 2.21% op/s
nativeObject(1000)            :        13291.83 +/- 1.19% op/s
nativeMap(1000)               :         6504.37 +/- 0.66% op/s
hashtrie(1000)                :            0.00 +/- 0.00% op/s
hamt(1000)                    :         5550.09 +/- 2.00% op/s
hamt_plus(1000)               :        35778.74 +/- 3.15% op/s
persistent-hash-trie(1000)    :         1173.81 +/- 0.93% op/s
mori hash_map(1000)           :         3424.90 +/- 2.18% op/s
immutable(1000)               :         9554.54 +/- 5.40% op/s
nativeObject(10000)           :          991.75 +/- 1.35% op/s
nativeMap(10000)              :          580.89 +/- 2.65% op/s
hashtrie(10000)               :            0.00 +/- 0.00% op/s
hamt(10000)                   :          579.33 +/- 1.63% op/s
hamt_plus(10000)              :         3970.65 +/- 8.14% op/s
persistent-hash-trie(10000)   :          138.95 +/- 1.84% op/s
mori hash_map(10000)          :          279.13 +/- 1.97% op/s
immutable(10000)              :         1194.37 +/- 2.52% op/s
```

File: benchmarks/remove_all.js [hunk 1]
```diff
@@ -14,6 +14,28 @@ var words = require('./words').words;
...
+    }
+  };
+};
+
+var nativeMapRemoveAll = function(keys, order) {
+  var h = api.nativeMapFrom(keys);
+  return function() {
+    var c;
+    for(var i = 0, len = order.length; i < len; ++i) {
+      c = new Map(h);
```
Comment (mattbierner) [inline benchmarks/remove_all.js]: These still need some work. It should be the cost to remove every element (without replacement) from a map. The current implementation starts with a new map every time. I believe it should be something like this:
```
var c =  new Map(h); // create full map at start
// now remove elements one at a time until all have been removed
for(var i = 0, len = order.length; i < len; ++i) {
    c = new Map(c); // copy the old map to make sure the original reference is not mutated.
    c.delete(order[i]);
}
```

Comment (blowery) [inline benchmarks/remove_all.js]: Fixed in 0d78590, good catch!

**Pull request merged**